src/groups/mqb/mqbs/mqbs_filestore.cpp

```diff
@@ -6479,7 +6476,11 @@ FileStore::generateReceipt(NodeContext*         nodeContext,
         nodeContext->d_state->unlock();
     }
     else {
-        bmqp::ProtocolUtil::buildReceipt(&nodeContext->d_blob,
+        // The pointer `nodeContext->d_blob_sp` might be in a write queue, so
+        // it's not safe to modify or replace the blob under this pointer.
+        // Instead, we get another shared pointer to another blob.
+        nodeContext->d_blob_sp = d_blobSpPool_p->getObject();
+        bmqp::ProtocolUtil::buildReceipt(nodeContext->d_blob_sp.get(),
```

Comment:
the `bmqp::ProtocolUtil::buildReceipt` can now assert fresh blob (no need for `removeAll`)